__init__.py

```diff
@@ -101,44 +101,68 @@ def __init__(self, bus):
-            if engine in blacklist:
-                continue
-            # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
-            plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
-            for config in configs:
-                d = {"plugin_name": plugin_display_name,
-                     "display_name": config.get("display_name", " "),
-                     "offline": config.get("offline", False),
-                     "lang": lang,
-                     "engine": engine}
-                stt_opts.append(d)
-                self._stt_opts[hash_dict(d)] = config
-        return stt_opts
-
-    def get_tts_lang_options(self, lang, blacklist=None):
-        blacklist = blacklist or []
-        tts_opts = []
-        cfgs = get_tts_lang_configs(lang=lang, include_dialects=True)
-        for engine, configs in cfgs.items():
-            if engine in blacklist:
-                continue
-            # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
-            plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
-            for voice in configs:
-                d = {"plugin_name": plugin_display_name,
-                     "display_name": voice.get("display_name", " "),
-                     "gender": voice.get("gender", " "),
-                     "offline": voice.get("offline", False),
-                     "lang": lang,
-                     'engine': engine}
-                tts_opts.append(d)
-                self._tts_opts[hash_dict(d)] = voice
-        return tts_opts
+    def get_stt_lang_options(self, lang, blacklist=None, preferred=None):
+        try:
+            blacklist = blacklist or []
+            stt_opts = []
+            cfgs = get_stt_lang_configs(lang=lang, include_dialects=True)
+            for engine, configs in cfgs.items():
+                if engine in blacklist:
+                    continue
+                # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
+                plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
+                for config in configs:
+                    d = {"plugin_name": plugin_display_name,
+                        "display_name": config.get("display_name", " "),
+                        "offline": config.get("offline", False),
+                        "lang": lang,
+                        "engine": engine}
+                    stt_opts.append(d)
+                    self._stt_opts[hash_dict(d)] = config
+
+                # Sort the list for GUI to display the preferred STT engine first
+                # allow images to set a preferred engine
+                if preferred and preferred not in blacklist:
+                    stt_opts = sorted(stt_opts, key=lambda x: x["engine"] != preferred, reverse=False)
```

Comment:
this does not look right, instead above replace `stt_opts.append(d)` with

```python
if engine == preferred:
     stt_opts.insert(0, d)  # start of list
else:
     stt_opts.append(d)
 ```

Reply:
removed